roboflow/models/instance_segmentation.py

```diff
@@ -0,0 +1,72 @@
+import io
+import urllib
+
+import requests
+from PIL import Image
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+from roboflow.config import INSTANCE_SEGMENTATION_MODEL, INSTANCE_SEGMENTATION_URL
+from roboflow.util.image_utils import validate_image_path
+from roboflow.util.prediction import PredictionGroup
+
+
+class InstanceSegmentationModel:
+    def __init__(
+        self,
+        api_key,
+        version_id,
+    ):
+        """
+        :param api_key: Your API key (obtained via your workspace API settings page)
+        :param version_id: The ID of the dataset version to use for predicting
+        """
+        self.__api_key = api_key
+        self.id = version_id
+
+        version_info = self.id.rsplit("/")
+        dataset_id = version_info[1]
+        version = version_info[2]
+
+        self.api_url = f"{INSTANCE_SEGMENTATION_URL}/{dataset_id}/{version}"
+
+    def predict(self, image_path):
+        """
+        Infers detections based on image from specified model and image path
+
+        :param image_path: Path to image (can be local or hosted)
+        :return: PredictionGroup --> a group of predictions based on Roboflow JSON response
+        """
+        validate_image_path(image_path)
+
+        params = {"api_key": self.__api_key}
+        request_kwargs = {}
+
+        hosted_image = urllib.parse.urlparse(image_path).scheme in (
+            "http",
+            "https",
+        )
+        if hosted_image:
+            params["image"] = image_path
+        else:
+            image = Image.open(image_path)
```

Comment:
other models seemed to open with `cv2` and then convert to `PIL`... i didn't understand that use-case, so i just used `PIL` here. if this makes sense we can refactor the others accordingly.